utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -0,0 +1,281 @@
+    val expectedNetworkBytesReceived = TrafficStats.getUidRxBytes(
+      ApplicationProvider.getApplicationContext<Application>().applicationInfo.uid
+    )
+
+    assertThat(performanceMetricsUtilsImpl.getTotalReceivedBytes())
+      .isEqualTo(expectedNetworkBytesReceived)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setAppProcesses_getMemoryUsage_returnsCorrectMemoryUsage() {
+    val activityManager: ActivityManager =
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    val process1 = ActivityManager.RunningAppProcessInfo().apply {
+      this.pid = TEST_PID
+      this.importanceReasonComponent = ComponentName("com.robolectric", "process 1")
+    }
+    val shadowActivityManager: ShadowActivityManager = shadowOf(activityManager)
+    shadowActivityManager.setProcesses(listOf(process1))
+    val processMemoryInfo = activityManager.getProcessMemoryInfo(arrayOf(TEST_PID).toIntArray())
+    val totalPssUsedTest = processMemoryInfo?.get(0)?.totalPss?.toLong() ?: 0L
+
+    assertThat(performanceMetricsUtilsImpl.getTotalPssUsed()).isEqualTo(totalPssUsedTest)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_removeCurrentApp_installTestApp_returnsCorrectApkSize() {
+    val application: Application = ApplicationProvider.getApplicationContext()
+    val applicationInfo = ApplicationInfo()
+    val testApkSize = (File(TEST_APP_PATH).length() / 1024)
+    applicationInfo.apply {
+      this.packageName = context.packageName
+      this.sourceDir = TEST_APP_PATH
+      this.name = TEST_PACKAGE_LABEL
+      this.flags = 0
+    }
+    val packageManager = application.packageManager
+    val shadowPackageManager = shadowOf(packageManager)
+
+    shadowPackageManager.removePackage(application.packageName)
+    shadowPackageManager.installPackage(
+      PackageInfo().apply {
+        this.packageName = context.packageName
+        this.applicationInfo = applicationInfo
+      }
+    )
+
+    val apkSize = performanceMetricsUtilsImpl.getApkSize()
+    assertThat(apkSize).isEqualTo(testApkSize)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  // TODO(#89): Move this to a common test application component.
+  @Module
+  class TestModule {
+
+    @Provides
+    @Singleton
+    fun providesContext(application: Application): Context = application
+
+    // TODO(#59): Either isolate these to their own shared test module, or use the real logging
+    // module in tests to avoid needing to specify these settings for tests.
+    @EnableConsoleLog
+    @Provides
+    fun provideEnableConsoleLog(): Boolean = true
+
+    @EnableFileLog
+    @Provides
+    fun provideEnableFileLog(): Boolean = false
+
+    @GlobalLogLevel
+    @Provides
+    fun provideGlobalLogLevel(): LogLevel = LogLevel.VERBOSE
+  }
+
+  @Module
+  class TestPlatformParameterModule {
```

Comment:
Could you maybe include the production parameter module instead of creating a test module like this? It doesn't seem like we'd care to change any of these values for the sake of testing the implementation.

Reply:
done.